src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,117 @@
+#include <aikido/constraint/SequentialSampleable.hpp>
+
+#include <dart/common/StlHelpers.hpp>
+
+namespace aikido {
+namespace constraint {
+
+// TODO (avk): Should I declare the class first and then define?
+// What is "good" C++ convention? Don't tell me it's a preference JS! -.-
```

Comment:
In that case, shall we add docstring during definitions, which we generally do not do right?

Reply:
> shall we add docstring during definitions

I think so. Although they are not exposed to the users, it would be very helpful for us (i.e., developers).